### DISCUSSION TITLE
how  datastream  use  scan.incremental.snapshot.enabled      not  sql

### DISCUSSION THREAD

**Discussion by X1304951263** [apache/flink-cdc#3232] [Q&A] [answered]
//debeziumProperties.setProperty("scan.incremental.snapshot.enabled", "true");  //无效

     //   Configuration flinkConfig = new Configuration();
     //   flinkConfig.setString("scan.incremental.snapshot.enabled", "true"); // 无效
     //   env.configure(flinkConfig);  //无效
     //   env.getConfig().setGlobalJobParameters(flinkConfig);   
这些方式 都不行，看大家 用的都是sql模式， datastream的 应该怎么使用呢，  用了以上几种方式  都 报错   User does not have the 'LOCK TABLES' privilege required to obtain a consistent snapshot by preventing concurrent writes to tables.

**Comment by yuxiqian** [comment, marked as answer]
Are you using `org.apache.flink.cdc.connectors.mysql.source.MySqlSource` package or the legacy `org.apache.flink.cdc.connectors.mysql.MySqlSource` one? By using the former one should be equivalent to passing `scan.incremental.snapshot.enabled = true`.

**Reply by X1304951263** [reply]
oh   i used   org.apache.flink.cdc.connectors.mysql.MySqlSource

**Reply by X1304951263** [reply]
Thank you very much for solving the problem that has been bothering me for a long time. Now, the vast majority of production issues need to be completed.
